pages/api/v1/secret/index.ts

```diff
@@ -0,0 +1,76 @@
+import { NextRequest, NextResponse } from "next/server";
+import { Redis } from "@upstash/redis";
+import { generateId } from "pkg/id";
+import { z } from "zod";
+
+export const requestValidation = z.object({
+  // ttl in seconds
+  // defaults to 30 days
+  // not more than 1 year
+  ttl: z
+    .string()
+    .nullable()
+    .transform((v) => (v ? parseInt(v, 10) : 43260))
+    .refine((v) => v > 0 && v <= 30758400, "ttl must be between 1 and 30758400 seconds"),
+
+  // number of reads before deletion
+  reads: z
+    .string()
+    .nullable()
+    .transform((v) => (v ? parseInt(v, 10) : null))
+    .refine((v) => v === null || v > 0, "reads must be greater than 0"),
```

Comment:
I think we need to refine this because it creates a bug: it allows you send null headers and the doesnt properly validate. 
I was able to create a null remainingReads secret and it never deletes.
![image](https://user-images.githubusercontent.com/16673873/213074614-a1852f55-b0ef-445f-9b8a-e8c7d2e64849.png)

But when sending with missing headers, it inserts an object with a null field which allows the
![image](https://user-images.githubusercontent.com/16673873/213074485-620cdbeb-1284-410a-9a5d-315c198f6268.png)

Reply:
If the intention is to have secrets that never expire until a timer runs out, then I can see that (but i just didnt see any note of that change or documentation for it so I wasnt sure if it was intentional 🤔 )

Reply:
I just quickly hacked this together last night, thanks for checking for bugs
I'll fix this

the 999 limit was just an arbitrary value in the frontend actually, not something with any real reason behind or set it stone :D

Reply:
Also I don't mind having secrets that never expire, that's probably quite useful, the default was just added as a default :D
so you have to actively set it to null, to have infinite retention
if you do not explicitely set it, then there will always be a finite retention

Reply:
I also have experience setting up docs pages. I was thinking of creating an API documentation page. But now that the API wont be exactly public (because its unencrypted) and instead rely on clients, I was thinking I'd hold off. I can get started on the docs on a branch using https://mintlify.com/ or https://docusaurus.io/